src/Http/Http.Extensions/src/RequestDelegateFactory.cs

```diff
@@ -1596,6 +1689,11 @@ private class FactoryContext
 
         public bool ReadForm { get; set; }
         public ParameterInfo? FirstFormRequestBodyParameter { get; set; }
+        // Properties for constructing and managing filters
+        public List<Expression> ContextArgAccess { get; } = new();
+        public Expression? MethodCall { get; set; }
+        public List<Expression> BoxedArgs { get; } = new();
```

Comment:
Is it possible to not allocate `ContextArgAccess` and `BoxedArgs` for endpoints with no parameters, or make them arrays and use `Array.Empty<Expression>` as the default instead?